packages/react/src/components/Table/TableHeader.tsx

```diff
@@ -11,45 +12,69 @@ interface TableHeaderProps extends ThHTMLAttributes<HTMLTableCellElement> {
   sortDirection?: SortDirection;
   onSort?: () => void;
   className?: string;
+  sortAscendingAnnouncement?: string;
+  sortDescendingAnnoucnement?: string;
 }
 
 const TableHeader = ({
   children,
   sortDirection,
   onSort,
   className,
+  sortAscendingAnnouncement = 'sorted ascending',
+  sortDescendingAnnoucnement = 'sorted descending',
   ...other
-}: TableHeaderProps) => (
-  <th
-    aria-sort={sortDirection}
-    className={classNames('TableHeader', className, {
-      'TableHeader--sort-ascending': sortDirection === 'ascending',
-      'TableHeader--sort-descending': sortDirection === 'descending'
-    })}
-    {...other}
-  >
-    {!!onSort && !!sortDirection ? (
-      <button
-        onClick={onSort}
-        className="TableHeader__sort-button"
-        type="button"
-      >
-        {children}
-        <span aria-hidden="true">
-          {sortDirection === 'none' ? (
-            <Icon type="sort-triangle" />
-          ) : sortDirection === 'ascending' ? (
-            <Icon type="triangle-up" />
-          ) : (
-            <Icon type="triangle-down" />
-          )}
-        </span>
-      </button>
-    ) : (
-      children
-    )}
-  </th>
-);
+}: TableHeaderProps) => {
+  // When the sort direction changes, we want to announce the change in a live region
+  // because changes to the sort value is not widely supported yet
+  // see: https://a11ysupport.io/tech/aria/aria-sort_attribute
+  const [announcement, setAnnouncement] = useState('');
+  useEffect(() => {
+    if (sortDirection !== 'none') {
+      setAnnouncement(
+        sortDirection === 'ascending'
+          ? sortAscendingAnnouncement
+          : sortDescendingAnnoucnement
+      );
+    }
+  }, [sortDirection]);
+  return (
+    <th
+      aria-sort={sortDirection}
+      className={classNames('TableHeader', className, {
+        'TableHeader--sort-ascending': sortDirection === 'ascending',
+        'TableHeader--sort-descending': sortDirection === 'descending'
+      })}
+      {...other}
+    >
+      {!!onSort && !!sortDirection ? (
+        <button
+          onClick={onSort}
+          className="TableHeader__sort-button"
+          type="button"
+        >
+          {children}
+          <span aria-hidden="true">
+            {sortDirection === 'none' ? (
+              <Icon type="sort-triangle" />
+            ) : sortDirection === 'ascending' ? (
+              <Icon type="triangle-up" />
+            ) : (
+              <Icon type="triangle-down" />
+            )}
+          </span>
+          <Offscreen>
+            <span role="alert" aria-live="polite">
```

Comment:
given our convo about the NVDA quirks that this patch addresses, do you think `role=status` would be better than `role=alert`?

Reply:
both inherit `aria-atomic=true` which is good

Reply:
Ah, yeah `role="status"` is probably a better attribute